cmake/FDBComponents.cmake

```diff
@@ -75,12 +59,57 @@ else()
   set(WITH_PYTHON OFF)
 endif()
 
+option(BUILD_PYTHON_BINDING "build python binding" ON)
+if(NOT BUILD_PYTHON_BINDING OR NOT WITH_PYTHON)
+  set(WITH_PYTHON_BINDING OFF)
+else()
+  if(WITH_PYTHON)
+    set(WITH_PYTHON_BINDING ON)
+  else()
+    #message(FATAL_ERROR "Could not found a suitable python interpreter")
+    set(WITH_PYTHON_BINDING OFF)
+  endif()
+endif()
+
+################################################################################
+# C Bindings
+################################################################################
+
+option(BUILD_C_BINDING "build C binding" ON)
+if(BUILD_C_BINDING AND WITH_PYTHON)
+  set(WITH_C_BINDING ON)
+else()
+  set(WITH_C_BINDING OFF)
+endif()
+
+################################################################################
+# Java Bindings
+################################################################################
+
+option(BUILD_JAVA_BINDING "build java binding" ON)
+if(NOT BUILD_JAVA_BINDING OR NOT WITH_C_BINDING)
+  set(WITH_JAVA_BINDING OFF)
```

Comment:
Maybe add a line

```
message(WARNING "Java depends on C binding, but C binding is not enabled")
```

or even promote WARNING to FATAL_ERROR since this might cause unexpected java binding missing?